user-service/src/main/java/org/java/mentorship/user/service/SessionService.java

```diff
@@ -0,0 +1,70 @@
+package org.java.mentorship.user.service;
+
+import lombok.AllArgsConstructor;
+import org.java.mentorship.contracts.user.dto.Session;
+import org.java.mentorship.contracts.user.dto.UserLoginRequest;
+import org.java.mentorship.user.domain.UserEntity;
+import org.java.mentorship.user.repository.SessionRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class SessionService {
+    private final SessionRepository sessionRepository;
+    private final UserService userService;
```

Comment:
This shouldn't have any logic for Users
It should call the user-service though a feignClient

Reply:
The problem is that the User service and SessionService both live on the same Spring app. I rather want to know if the way I implemented sessions is secure